src/libical/astime.h

```diff
@@ -46,24 +71,18 @@
 
 typedef struct ut_instant
 {
-    double j_date;      /**< julian decimal date, 0 = 01 Jan 4713 BC 12 HR UT */
-    long year;          /**< year, valid range [-4,713, +2,147,483,647] */
+    long j_date0;      /**< julian decimal date, 0 = 01 Jan 4713 BC */
+    long year;          /**< year, valid range [-4,713, +32,767] */
     int month;          /**<    [1-12]  */
     int day;            /**<    [1-31]  */
-    int i_hour;         /**<    [0-23]  */
-    int i_minute;               /**<    [0-59]  */
-    int i_second;               /**<    [0-59]  */
-    double d_hour;              /**< [0.0-23.9999] includes minute and second */
-    double d_minute;            /**<    [0.0-59.9999] includes second   */
-    double d_second;            /**<    [0.0-59.9999]   */
     int weekday;                /**<    [0-6]   */
     int day_of_year;            /**<    [1-366] */
 } UTinstant, *UTinstantPtr;
 
 /*      Functions in caldate.c  */
 
-LIBICAL_ICAL_EXPORT long caldat(UTinstantPtr);      /** converts julian date to year,mo,da */
+LIBICAL_ICAL_EXPORT void caldat(UTinstantPtr);    /** converts julian date to year,mo,da */
```

Comment:
I don't want to break binary and source compatibility.
we can add new new functions and deprecate the old one, however.

so I guess let's continue to return a value or we need a function.

Reply:
right.  if we don't need to export the new ones (hopefully not) then let's not.

Reply:
put the doxygen comment like the following in astime.h immediately above the function declaration.  for example:
/** Converts Julian date to year,mo, da
 ** @deprecated use caldat_int() instead 
*/
 LIBICAL_ICAL_EXPORT long caldat(UTinstantPtr); 

Also let's mention the deprecated functions in the top-level ReleaseNotes.txt.
Do we want this in a 3.0.5 release or is master (3.1.0) good enought?

Reply:
Updated as suggested. Not sure, whether the new funcs should be mentioned in the deprecation message, as they are considered internal now (not exported). Still mentioned them to clearify internal use.

I think, 3.1.0 should be fine.